File: java/v1/src/main/java/com/api/v1/dtos/requests/BookRegistrationRequestDto.java
package com.api.v1.dtos.requests;

import com.api.v1.annotations.ISBN;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public record BookRegistrationRequestDto(
        @NotBlank String title,
        String subtitle,
        @ISBN String isbn,
        @NotBlank String author,
        @NotBlank String field,
        int numberOfPages,
        @Min(1) int version,
        double price,
        @NotBlank String publisher,
        int publishingYear
) {
}

File: kotlin/v2/src/main/kotlin/com/api/v2/book/dtos/BookRegistrationRequestDto.kt
package com.api.v2.book.dtos

import com.api.v2.book.annotations.ISBN
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank

data class BookRegistrationRequestDto(
    val title: @NotBlank String,
    val subtitle: String?,
    val isbn: @ISBN String,
    val author: @NotBlank String,
    val field: @NotBlank String,
    val publisher: @NotBlank String,
    val numberOfPages: @Min(1) Int,
    val version: @Min(1) Int,
    val publishingYear: Int,
    val price: Double
)
